genie-agent/src/test/groovy/com/netflix/genie/agent/execution/services/impl/FileSystemFetchingCacheServiceSpec.groovy

```diff
@@ -0,0 +1,287 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services.impl
+
+import com.netflix.genie.agent.cli.ArgumentDelegates
+import com.netflix.genie.agent.execution.exceptions.DownloadException
+import com.netflix.genie.agent.execution.locks.impl.LockingFactory
+import com.netflix.genie.test.categories.UnitTest
+import org.apache.commons.lang3.tuple.Pair
+import org.assertj.core.util.Sets
+import org.junit.Rule
+import org.junit.experimental.categories.Category
+import org.junit.rules.TemporaryFolder
+import org.springframework.core.io.Resource
+import org.springframework.core.io.ResourceLoader
+import spock.lang.Specification
+
+import java.nio.charset.StandardCharsets
+
+@Category(UnitTest.class)
+class FileSystemFetchingCacheServiceSpec extends Specification {
+
+    ResourceLoader resourceLoader
+    ArgumentDelegates.CacheArguments cacheArguments
+    FileSystemFetchingCacheService cache
+    Resource resource
+    long DEFAULT_RESOURCE_LAST_MODIFIED_TS = 100
+    URI uri
+    LockingFactory lockingFactory;
+
+    @Rule
+    TemporaryFolder temporaryFolder
+
+    void setup() {
+        temporaryFolder.create()
+        resourceLoader = Mock()
+        cacheArguments = Mock()
+        cacheArguments.getCacheDirectory() >> temporaryFolder.getRoot()
+        resource = Mock()
+        lockingFactory = new LockingFactory();
+        resourceLoader.getResource(_ as String) >> resource
```

Comment:
Yup, aware of that. However, since `resourceLoader.getResource(_ as String)` was always called when cache.get was called I had not copied it everywhere as I was not going to check the number of its invocations. I was treating this more as a one time set up. Same is true for `cacheArguments.getCacheDirectory()`. Just for purity purposes, I have copied `1*resourceLoader.getResource(_ as String)` in all the tests and removed it from set up but have left   `cacheArguments.getCacheDirectory()` in set up as its just a set up of the top level directory. Have also created LockFactory and Lock mocks as per your other comment. But have created it once in the set up block